streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -504,9 +593,18 @@ private void connectProcessorAndStateStore(String processorName, String stateSto
     public Map<Integer, TopicsInfo> topicGroups(String applicationId) {
         Map<Integer, TopicsInfo> topicGroups = new HashMap<>();
 
+
```

Comment:
nit: add some comments explaining about why we need to update the `nodeToSourceTopics` here.